public/components/trace_analytics/components/traces/span_detail_panel.tsx

```diff
@@ -135,72 +135,98 @@ export function SpanDetailPanel(props: {
     }
   };
 
-  const refresh = debounce(() => {
-    if (isEmpty(props.colorMap)) return;
-    const refreshDSL = spanFiltersToDSL();
-    setDSL(refreshDSL);
-    handleSpansGanttRequest(
-      props.traceId,
-      props.http,
-      setData,
-      props.colorMap,
-      refreshDSL,
-      mode,
-      props.dataSourceMDSId
-    ).finally(() => setIsGanttChartLoading(false));
-  }, 150);
-
-  const spanFiltersToDSL = () => {
-    const spanDSL: any =
-      mode === 'jaeger'
-        ? {
-            query: {
-              bool: {
-                must: [
-                  {
-                    term: {
-                      traceID: props.traceId,
-                    },
-                  },
-                ],
-                filter: [],
-                should: [],
-                must_not: [],
-              },
-            },
+  const parseAndFilterHits = (
+    payloadData: string,
+    traceMode: string,
+    payloadSpanFilters: any[]
+  ) => {
+    try {
+      const parsed = JSON.parse(payloadData);
+      let hits: any[] = [];
+      if (parsed.hits && Array.isArray(parsed.hits.hits)) {
+        hits = parsed.hits.hits;
+      } else if (Array.isArray(parsed)) {
+        hits = parsed;
+      } else {
+        console.warn('Unexpected payload format:', parsed);
+        return [];
+      }
+
+      hits = hits.map((hit) => {
+        if (traceMode === 'jaeger') {
+          if (!hit.sort || !hit.sort[0]) {
+            return {
+              ...hit,
+              sort: [Number(hit._source.startTime) * 1000], // Jaeger: startTime is in microseconds
+            };
           }
-        : {
-            query: {
-              bool: {
-                must: [
-                  {
-                    term: {
-                      traceId: props.traceId,
-                    },
-                  },
-                ],
-                filter: [],
-                should: [],
-                must_not: [],
-              },
-            },
-          };
-    spanFilters.map(({ field, value }) => {
-      if (value != null) {
-        spanDSL.query.bool.filter.push({
-          term: {
-            [field]: value,
-          },
+        } else {
+          if (!hit.sort || !hit.sort[0]) {
```

Comment:
nit. since this if is the same for both modes, it might be better to check `traceMode` inside this if

Reply:
Updated too 
```
      hits = hits.map((hit) => {
        if (!hit.sort || !hit.sort[0]) {
          const time = traceMode === 'jaeger' 
            ? Number(hit._source.startTime) * MILI_TO_SEC
            : parseIsoToNano(hit._source.startTime);
          
          return {
            ...hit,
            sort: [time],
          };
        }
        return hit;
      });
```